src/matrix/layout_info.cpp

```diff
@@ -22,45 +22,36 @@ LayoutInfo::LayoutInfo(const LocalElementSize& size, const TileElementSize& bloc
   using util::size_t::sum;
   using util::size_t::mul;
 
-  if (!size_.isValid()) {
-    throw std::invalid_argument("Error: Invalid Matrix size");
-  }
-  if (!block_size_.isValid() || block_size_.isEmpty()) {
-    throw std::invalid_argument("Error: Invalid Block size");
-  }
+  DLAF_ASSERT(size_.isValid(), "Invalid Matrix size!");
+  DLAF_ASSERT(block_size_.isValid() && !block_size_.isEmpty(), "Invalid Block size!");
 
   nr_tiles_ = {util::ceilDiv(size_.rows(), block_size_.rows()),
                util::ceilDiv(size_.cols(), block_size_.cols())};
 
-  if (size_.isEmpty()) {
-    if (ld_tile_ < 1) {
-      throw std::invalid_argument("Error: Invalid Leading Dimension");
-    }
-    if (tile_offset_row_ < 1) {
-      throw std::invalid_argument("Error: Invalid Tile Row Offset");
-    }
-    if (tile_offset_col_ < 1) {
-      throw std::invalid_argument("Error: Invalid Tile Col Offset");
-    }
+  //  DLAF_ASSERT((!size_.isEmpty() || ld_tile_ >= 1), "Invalid Leading Dimension");
+  //  DLAF_ASSERT((!size_.isEmpty() || tile_offset_row_ >= 1), "Invalid Tile Row Offset");
+  //  DLAF_ASSERT((!size_.isEmpty() || tile_offset_col_ >= 1), "Invalid Tile Col Offset");
+  if (!size_.isEmpty()) {
```

Comment:
I think it is the opposite check...isn't it?

Reply:
true 😅